File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/RouteTileDownloadListener.java
package com.mapbox.services.android.navigation.v5.navigation;

import android.support.annotation.NonNull;

/**
 * Listener for receiving updates about a route tile download.
 */
public interface RouteTileDownloadListener {

  /**
   * Called if there is an error with the downloading.
   *
   * @param error with message description
   */
  void onError(@NonNull OfflineError error);

  /**
   * Called with percentage progress updates of the download.
   *
   * @param percent completed
   */
  void onProgressUpdate(int percent);

  /**
   * Called when download was completed.
   */
  void onCompletion();
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/RouteTileDownloadListener.kt
package com.mapbox.services.android.navigation.v5.navigation

/**
 * Listener for receiving updates about a route tile download.
 */
interface RouteTileDownloadListener {

    /**
     * Called if there is an error with the downloading.
     *
     * @param error with message description
     */
    fun onError(error: OfflineError)

    /**
     * Called with percentage progress updates of the download.
     *
     * @param percent completed
     */
    fun onProgressUpdate(percent: Int)

    /**
     * Called when download was completed.
     */
    fun onCompletion()
}
